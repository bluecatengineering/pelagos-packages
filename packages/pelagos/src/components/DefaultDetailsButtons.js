import {useCallback} from 'react';
import PropTypes from 'prop-types';
import {t} from '@bluecateng/l10n.macro';

import Button from './Button';

/** Default buttons for the details panel. */
const DefaultDetailsButtons = ({className, item, disableEdit, disableDelete, onEdit, onDelete}) => {
	const handleEdit = useCallback(() => onEdit(item), [item, onEdit]);
	const handleDelete = useCallback(() => onDelete(item), [item, onDelete]);
	return (
		<div className={className}>
			{onDelete && (
				<Button
					id="deleteBtn"
					text={t`Delete`}
					tooltipText={disableDelete}
					disabled={!!disableDelete}
					onClick={handleDelete}
				/>
			)}
			<Button
				id="editBtn"
				text={t`Edit`}
				tooltipText={disableEdit}
				disabled={!!disableEdit}
				type="primary"
				onClick={handleEdit}
			/>
		</div>
	);
};

DefaultDetailsButtons.propTypes = {
	/** The component class name(s). */
	className: PropTypes.string,
	/** The item to display, it must have at least two properties: id and name  */
	item: PropTypes.object.isRequired,
	/** If set the edit button is disabled and this text displayed as tooltip. */
	disableEdit: PropTypes.string,
	/** If set the delete button is disabled and this text displayed as tooltip. */
	disableDelete: PropTypes.string,
	/** Function invoked when the edit button is clicked. */
	onEdit: PropTypes.func,
	/** Function invoked when the delete button is clicked. */
	onDelete: PropTypes.func,
};

export default DefaultDetailsButtons;
