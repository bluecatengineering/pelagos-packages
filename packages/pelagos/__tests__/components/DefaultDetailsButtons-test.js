import {shallow} from 'enzyme';

import DefaultDetailsButtons from '../../src/components/DefaultDetailsButtons';

jest.unmock('../../src/components/DefaultDetailsButtons');

const item = {name: 'Test'};

describe('DefaultDetailsButtons', () => {
	describe('rendering', () => {
		it('renders expected elements', () => {
			const wrapper = shallow(
				<DefaultDetailsButtons className="TestClass" item={item} onEdit={jest.fn()} onDelete={jest.fn()} />
			);
			expect(wrapper.getElement()).toMatchSnapshot();
		});

		it('renders expected elements when onDelete is not set', () => {
			const wrapper = shallow(<DefaultDetailsButtons item={item} onEdit={jest.fn()} />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});

		it('renders expected elements when disableEdit and disableDelete are set', () => {
			const wrapper = shallow(
				<DefaultDetailsButtons
					item={item}
					disableEdit="Disable Edit"
					disableDelete="Disable Delete"
					onEdit={jest.fn()}
					onDelete={jest.fn()}
				/>
			);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});

	describe('behaviour', () => {
		it('calls onEdit when the edit button is clicked', () => {
			const onEdit = jest.fn();
			const wrapper = shallow(<DefaultDetailsButtons item={item} onEdit={onEdit} />);
			wrapper.find('#editBtn').prop('onClick')();
			expect(onEdit).toHaveBeenCalled();
		});

		it('calls onDelete when the delete button is clicked', () => {
			const onDelete = jest.fn();
			const wrapper = shallow(<DefaultDetailsButtons item={item} onDelete={onDelete} />);
			wrapper.find('#deleteBtn').prop('onClick')();
			expect(onDelete).toHaveBeenCalled();
		});
	});
});
