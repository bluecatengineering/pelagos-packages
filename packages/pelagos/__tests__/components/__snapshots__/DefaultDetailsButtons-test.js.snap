// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultDetailsButtons rendering renders expected elements 1`] = `
<div
  className="TestClass"
>
  <Button
    disabled={false}
    id="deleteBtn"
    onClick={[Function]}
    size="medium"
    text="Delete"
    type="tertiary"
  />
  <Button
    disabled={false}
    id="editBtn"
    onClick={[Function]}
    size="medium"
    text="Edit"
    type="primary"
  />
</div>
`;

exports[`DefaultDetailsButtons rendering renders expected elements when disableEdit and disableDelete are set 1`] = `
<div>
  <Button
    disabled={true}
    id="deleteBtn"
    onClick={[Function]}
    size="medium"
    text="Delete"
    tooltipText="Disable Delete"
    type="tertiary"
  />
  <Button
    disabled={true}
    id="editBtn"
    onClick={[Function]}
    size="medium"
    text="Edit"
    tooltipText="Disable Edit"
    type="primary"
  />
</div>
`;

exports[`DefaultDetailsButtons rendering renders expected elements when onDelete is not set 1`] = `
<div>
  <Button
    disabled={false}
    id="editBtn"
    onClick={[Function]}
    size="medium"
    text="Edit"
    type="primary"
  />
</div>
`;
